core/modules.rs

```diff
@@ -806,6 +810,122 @@ pub(crate) struct ModuleMap {
 }
 
 impl ModuleMap {
+  pub fn to_v8_object(
```

Comment:
Rename to `serialize_for_snapshotting` - ideally we would return a vector of `v8::Data` that can just be iterated over and directly added using `scope.add_context_data()`